### PR TITLE
initial containers.conf.d work

### PR DESCRIPTION
**Pull request by maxamillion**
Signed-off-by: Adam Miller <admiller@redhat.com>

### CODE REVIEW

**Comment by richm** [comment]
you can run a local yamllint check with `tox -e yamllint`

**Comment by maxamillion** [comment]
> you can run a local yamllint check with `tox -e yamllint`

fixed! :)

File: tests/tests_basic.yml [hunk 1]
```diff
@@ -36,3 +48,29 @@
     - name: Run role again to test for idempotency
       include_role:
         name: linux-system-roles.podman
+
+    - name: Run Assertion Tests - System
+      when: podman_run_as_user == "root"
+      block:
+        - name: state containers.conf.d
```
Comment (richm) [inline tests/tests_basic.yml:55]: stat

File: tasks/handle_container_conf_d.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+
+- name: Ensure containers.d exists - system
+  file:
+    path: "{{ __podman_containers_conf_system|dirname }}"
```
Comment (richm) [inline tasks/handle_container_conf_d.yml:5]: Something weird is happening with this macro eval - there is a double quote being added to the path!
```
    "path": "\"/etc/containers/containers.conf.d",
```
which I think is why the test fails - see also
```
    "dest": "\"/etc/containers/containers.conf.d/50-systemroles.conf\"\n",
```
which has extra leading and trailing double quotes (and newline?) - maybe the `dirname` filter is errant?

Comment (maxamillion) [reply tasks/handle_container_conf_d.yml:5]: oh goodness ... that's fun, when I run the tox tests locally it worked fine on Fedora 36 ... alright, I'll dig into that a bit. Thanks!

**Comment by richm** [comment]
[citest bad]

File: tasks/handle_container_conf_d.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    state: "directory"
```
Comment (richm) [inline tasks/handle_container_conf_d.yml:6]: ansible-lint says you must have `owner`, `group`, and `mode` for files/directories created by the role.

**Pull request merged**